crates/core/src/operations/clone.rs

```diff
@@ -0,0 +1,222 @@
+use crate::operations::create::CreateBuilder;
+use crate::protocol::{DeltaOperation, SaveMode};
+use crate::table::builder::DeltaTableBuilder;
+use crate::{DeltaResult, DeltaTable, DeltaTableError};
+
+pub async fn shallow_clone(
+    source: Url,
+    target: Url,
+    version: Option<i64>,
+) -> DeltaResult<DeltaTable> {
+    // Validate that source and target are both filesystem Urls. If not, return an error.
+    if source.scheme() != "file" || target.scheme() != "file" {
+        return Err(DeltaTableError::InvalidTableLocation(format!(
+            "shallow_clone() requires file:// URLs for both source and target; got source='{}' (scheme='{}'), target='{}' (scheme='{}')",
+            source,
+            source.scheme(),
+            target,
+            target.scheme()
+        )));
+    }
+
+    // 1) Load source table and snapshot
+    let mut src_table = DeltaTableBuilder::from_uri(source)?.load().await?;
+    if let Some(v) = version {
+        src_table.load_version(v).await?;
+    }
+    let src_state = src_table.snapshot()?;
+    let src_snapshot: &EagerSnapshot = src_state.snapshot();
+    let src_metadata = src_state.metadata().clone();
+    let src_schema = src_metadata.parse_schema()?;
+    let partition_columns = src_metadata.partition_columns().to_vec();
+    let configuration = src_metadata.configuration().clone();
+    let src_protocol = src_state.protocol().clone();
+    let src_log = src_table.log_store();
+
+    // 2) Create a target table mirroring source metadata and protocol
+    let mut create = CreateBuilder::new()
+        .with_location(target.as_ref().to_string())
+        .with_columns(src_schema.fields().cloned())
+        .with_partition_columns(partition_columns.clone())
+        .with_configuration(
+            configuration
+                .iter()
+                .map(|(k, v)| (k.clone(), Some(v.clone()))),
+        );
+
+    if let Some(n) = src_metadata.name() {
+        create = create.with_table_name(n.to_string());
+    }
+    if let Some(d) = src_metadata.description() {
+        create = create.with_comment(d.to_string());
+    }
+
+    let mut target_table = create
+        .with_actions([Action::Protocol(src_protocol.clone())])
+        .await?;
+
+    // 3) Gather source files from snapshot
+    let file_views: Vec<_> = src_snapshot
+        .file_views(&src_log, None)
+        .try_collect()
+        .await?;
+
+    let mut actions = Vec::with_capacity(file_views.len() + 1);
+    actions.push(Action::Metadata(src_metadata));
+
+    // 4) Add files to the target table
+    let target_root_path = target_table
+        .log_store()
+        .config()
+        .location
+        .to_file_path()
+        .unwrap();
+    let source_root_path = src_log.config().location.to_file_path().unwrap();
+
+    for view in file_views {
+        let mut add = view.add_action();
+        add.data_change = true;
+        // Absolute paths are not supported for now, create symlinks instead.
```

Comment:
I think we should address this first. As we also want compatibility with shallow clones that have been created by the Java library and use absolute paths. Based on this comment I was under the impression that this was close to working: https://github.com/delta-io/delta-rs/issues/2456#issuecomment-3127730975

Reply:
My impression is this might be quite far away. I tried this first with absolute paths and it did not work. The code prepends the directory path to the absolute path giving nonsense. So, I think we need to ask the maintainers because this might still be far away. The good news is that most of the code can remain the same once they add this.

Reply:
Hopefully we could fix this ourselves rather than needing to wait for them to add it? Usually most path libraries will handle `join(absolute_directory_path, absolute_path)` by just returning the second path, maybe there's some tidying up of their path/URL handling that's needed.